skore/src/skore/sklearn/find_ml_task.py

```diff
@@ -10,20 +10,25 @@
 
 def _is_sequential(y) -> bool:
     """Check whether ``y`` is vector of sequential integer values."""
-    y_values = np.sort(np.unique(y))
+    y_values = np.unique(y)
     sequential = np.arange(y_values[0], y_values[-1] + 1)
     return np.array_equal(y_values, sequential)
 
 
 def _is_classification(y) -> bool:
     """Determine if `y` is a target for a classification task.
 
-    If `y` contains integers, sklearn's `type_of_target` considers
-    the task to be multiclass classification.
-    This function makes the analysis finer.
+    If `y` contains integers, sklearn's `type_of_target` considers the task
+    to be multiclass classification. This might not be the case, so we add the
+    constraints that `y` must contain sequential values and contain 0.
+
+    If `y` does not contain numbers (e.g. strings), then this function returns True.
     """
-    y = y.flatten()
-    return _is_sequential(y) and 0 in y
+    y = np.array(y).flatten()
```

Comment:
I'm thinking that we could use `sklearn.utils.check_array` that should help.

```suggestion
    y = check_array(y, dtype="numeric", ensure_2d=False)
```

Whatever `ValueError` should be related to the type not being "numeric"

Reply:
I don't mind `y` containing strings, in this case we know that the task is classification

Reply:
we use check_array now